funnel/models/event.py

```diff
@@ -124,6 +160,29 @@ class TicketClient(BaseMixin, db.Model):
     proposal_space = db.relationship(ProposalSpace,
         backref=db.backref('ticket_clients', cascade='all, delete-orphan'))
 
+    def import_from_list(self, space, ticket_list, cancel_list=[]):
+        """
+        Batch upserts the tickets and its associated ticket types and participants.
+        Cancels the tickets in cancel_list.
+        """
+        for ticket_dict in ticket_list:
+            ticket_type = TicketType.upsert(space, TicketType.get_name(space, ticket_dict['ticket_type']),
+                            title=ticket_dict['ticket_type'], proposal_space=space)
```

Comment:
This is an anti-pattern. You shouldn't be calling `TicketType.<method>` twice in the same call. If `title` is so important to this class, change the class to index `title`, and change `get` and `upsert` to accept either `name` or `title` as an identifier for the row.